pkg/ccl/streamingccl/streamingest/alter_replication_job.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package streamingest
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/colinfo"
+	"github.com/cockroachdb/cockroach/pkg/sql/exprutil"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/cockroachdb/errors"
+)
+
+const alterReplicationJobOp = "ALTER TENANT REPLICATION"
+
+var alterReplicationJobHeader = colinfo.ResultColumns{
+	{Name: "replication_job_id", Typ: types.Int},
+}
+
+func alterReplicationJobTypeCheck(
+	ctx context.Context, stmt tree.Statement, p sql.PlanHookState,
+) (matched bool, header colinfo.ResultColumns, _ error) {
+	alterStmt, ok := stmt.(*tree.AlterTenantReplication)
+	if !ok {
+		return false, nil, nil
+	}
+	if err := exprutil.TypeCheck(
+		ctx, alterReplicationJobOp, p.SemaCtx(), exprutil.Strings{alterStmt.TenantName},
+	); err != nil {
+		return false, nil, err
+	}
+	return true, alterReplicationJobHeader, nil
+}
+
+func alterReplicationJobHook(
+	ctx context.Context, stmt tree.Statement, p sql.PlanHookState,
+) (sql.PlanHookRowFn, colinfo.ResultColumns, []sql.PlanNode, bool, error) {
+	alterTenantStmt, ok := stmt.(*tree.AlterTenantReplication)
+	if !ok {
+		return nil, nil, nil, false, nil
+	}
+	exprEval := p.ExprEvaluator(alterReplicationJobOp)
+	name, err := exprEval.String(ctx, alterTenantStmt.TenantName)
+	if err != nil {
+		return nil, nil, nil, false, err
+	}
+	tenantName := roachpb.TenantName(name)
+
+	fn := func(ctx context.Context, _ []sql.PlanNode, resultsCh chan<- tree.Datums) error {
+		tenInfo, err := sql.GetTenantRecordByName(ctx, p.ExecCfg(), p.Txn(), tenantName)
```

Comment:
`GetTenantRecordByName()` doesn't do a system tenant check or an admin check.  We might want to add those above here so that we get a good error message.

We should probably clean up how some of this is factored in a follow-up PR.

Reply:
good point, added a system tenant check as well as an enterprise check.